src/iuse.cpp

```diff
@@ -1841,6 +1841,31 @@ int iuse::purify_iv(player *p, item *it, bool, const tripoint& )
     return it->type->charges_to_use();
 }
 
+void spawn_spores( player *p ) {
+    int spore_spawned = 0;
+    for ( int x = p->posx() - 4; x <= p->posx() + 4; x++ ) {
+        for ( int y = p->posy() - 4; y <= p->posy() + 4; y++ ) {
+            tripoint dest( x, y, p->posz() );
+            if ( rng( 0, 10 ) > trig_dist( x, y, p->posx(), p->posy() ) &&
+                 rng( 0, 10 ) > trig_dist( x, y, p->posx(), p->posy() ) ) {
```

Comment:
This check is weird. Why not save the distance instead? It is later used in monster spawning check anyway.
What's the deal with double compare against random values? Not sure what author wanted to achieve, but there are cleaner ways of doing it.